corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -209,6 +210,25 @@ FormplayerFrontend.on("start", function (options) {
     }
 });
 
+FormplayerFrontend.reqres.setHandler('getCurrentAppId', function() {
+    // First attempt to grab app id from URL
+    var urlObject = Util.currentUrlToObject(),
+        user = FormplayerFrontend.request('currentUser'),
+        appId;
+
+    appId = urlObject.appId;
+
+    if (appId) {
+        return appId;
+    }
+
+    // If it's not in the URL, then we are either on the home screen of formplayer
+    // and there is no app selected, or we are in preview mode.
+    appId = user.previewAppId;
```

Comment:
Might be time to write a new front end! Man technology advances so fast these days...